### ISSUE TITLE
Temperature& humidity sensor battery level

### ISSUE THREAD

**Issue by mateutek** [aholstenson/miio#74]
Hi,
I was wondering if there is a way to get the temp sensor power lvl. I saw in few different projects that they were able to get the data for battery lvl, but i couldn't reverse eng it in miio.
Does anyone have an idea on how to approach this? I am starting to work with wireshark, but it will take some time to get any results.

**Comment by mirko3000** [comment]
In the raw data you can see a property names "voltage", that should somehow give you a kind of battery level. My sensors return values between 2915 and 3015, they are still pretty new. Don't know how the voltage level behaves in time...

**Comment by realhoffi** [comment]
this one reads the voltage, but it sends the cmd "read".

https://github.com/marvinroger/node-lumi-aqara

i thinkyou can't read it with "get_prop" or "get_device_prop_exp"

**Comment by mateutek** [comment]
@mirko3000 Where do you access and read raw data?? Could't find anything in the code

**Comment by mirko3000** [comment]
@mateutek I found out by checking the data in subdevice.js, `_report(data)`. If you want to use the voltage property you first need to define it in your code by calling 
`myDevice.defineProperty('voltage');`
Then you will receive change events on the voltage property. Note that the property is initially not set, so you need to wait for the first event.

**Comment by mateutek** [comment]
@mirko3000 Thanks, i figured it out after i posted the comment ;)
But now i tried to load the properties before event emit with this code,so i can have initial value.I might consider using some DB to save this data, but i want to have control over loading this properties.
```javascript
let promise = device.call('get_prop',['temperature','humidity','voltage']);
```
I get 
```bash
{ Error: Method not found.
    at Object.reject (.../node_modules/miio/lib/device.js:217:13)
    at Gateway._onMessage (.../node_modules/miio/lib/device.js:108:8)
    at emitTwo (events.js:106:13)
    at Socket.emit (events.js:191:7)
    at UDP.onMessage (dgram.js:548:8) code: -32601 }
```
It works like a charm for normal device.
```javascript
let promise = device.call('get_prop',['power','aqi','mode', 'motor1_speed']);
```
```
[ 'on', 35, 'silent', 346 ]
```

**Comment by aholstenson** [comment]
`call` provides raw, unfiltered access to the API used internally by Xiaomi. Subdevices connected to a gateway do not share the same API as other devices. You can find the raw API-names in the file `device.js` located in the `gateway` folder.

You can also try using `loadProperties` and `getProperties`.

**Comment by mateutek** [comment]
Is there a way to force get the battery lvl ?? Even via gateway ??

**Comment by realhoffi** [comment]
@mateutek i think @aholstenson has to do something! in the link i posted, the query returns a battery level!

but you can define a property "voltage" on watch für the "propertyChange" event! this works

**Comment by mateutek** [comment]
@realhoffi Yeah, i saw that i got some value, but when i restart the app it gets down to 0, so i want to force get it. But on the other hand i want some chart with the levels so in the end i will get some DB
to save the data

**Comment by aholstenson** [comment]
I've looked a bit at this and there does not seem to be a way to read the voltage or battery level without using the Developer API for reads. Using the Developer API for reads has caused issues for some users that also use other libraries to interact with the Aqara gateway.

I'm leaning towards adding support for this via the Developer API despite that, it's a great feature to have in the library.

**Comment by aholstenson** [comment]
Support for this is now available in 0.15.3.
